blog/2022-11-01-cfd-protocol-explained.md

```diff
@@ -0,0 +1,247 @@
+
+Luckily, this kind of problem has been solved in the past, the most prominent example being [the Lightning Network's revocation mechanism](https://github.com/lightning/bolts/blob/93909f67f6a48ee3f155a6224c182e612dd5f187/05-onchain.md#revoked-transaction-close-handling).
+Having worked on an [implementation](https://github.com/comit-network/thor/) of [Generalized Bitcoin-Compatible Channels](https://eprint.iacr.org/2020/476.pdf), we knew that we wanted to apply those same constructs to our CFD protocol.
+
+##### Commitment transaction
+
+The `lock_transaction`'s output is now fittingly referred to as the `lock_output`.
+Its purpose is only to certify the creation of the CFD.
+After its publication parties can no longer back out of the contract.
+Its output is no longer spent by CETs directly.
+Instead, a `commitment_transaction` is introduced, which spends the `lock_output` into the `dlc_output`.
+
+The `dlc_output`, now part of a different transaction in the protocol, is still spent in the same way described previously: a set of CETs define how the `dlc_output` is split up based on the price of BTC in USD.
+The purpose of the `commitment_transaction` is to enforce that the `dlc_output` can only be spent by the CETs which were constructed together with it.
+
+Importantly, the `refund_transaction` works the same as before: it spends the `dlc_output`, which is now an output to the `commitment_transaction`.
+We focus on the CETs for clarity, but everything applies to the `refund_transaction` as was discussed in previous sections.
+
+As an example, we can imagine maker and taker agreeing to roll over the CFD because they're both optimistic that the price movement of BTC will benefit them if they wait longer.
+During `contract_setup`, they created and signed `commitment_transaction_0` and a whole set of CETs along with it.
+During `rollover`, they produce a new `commitment_transaction_1` and a new collection of CETs based on a new `announcement` from `olivia`.
+Since they have the means to publish fully signed `commitment_transaction`s, they can eventually publish `commitment_transaction_1` and settle the CFD based on a newer `attestation` from `olivia`.
+
+##### Punish transaction
+
+But, what happens if either party decides to publish an old `commitment_transaction_0` and settle the CFD based on an old BTC price attested by the oracle?
+That would defeat the purpose of rolling over the CFD.
+This is where the revocation mechanism comes in.
+
+A `commitment_transaction`'s `dlc_output` can actually be spent in one more way.
+In full, the `commitment_transaction` can be spent with:
+
+    1. A series of CETs unlocked by one of `olivia`'s `attestation`s.
+    2. A `refund_transaction` that can only be published at least some time after `olivia` has published the relevant `attestation`.
```

Comment:
This sounds like that the refund transaction depends on the actual attestation. 
Can we rephrase this please? Maybe this: 

```suggestion
    2. A `refund_transaction` that can only be published at least some time after `olivia` should have published the relevant `attestation`.
```

Reply:
Since this sentence is just a recap I've chosen to modify the content and not just the style.